server/src/main/java/com/epam/aidial/core/server/service/ApplicationService.java

```diff
@@ -208,6 +223,23 @@ public List<Application> getApplications(ResourceDescriptor resource, Consumer<R
         return applications;
     }
 
+    public void validateCustomApplication(Application application, ProxyContext context) {
+        try {
+            Config config = context.getConfig();
+            List<ResourceDescriptor> files = CustomApplicationUtils.getFiles(config, application, encryptionService,
+                    resourceService);
+            files.stream().filter(resource -> !(context.getProxy().getAccessService().hasReadAccess(resource, context)))
+                    .findAny().ifPresent(file -> {
+                        throw new HttpException(BAD_REQUEST, "No read access to file: " + file.getUrl());
+                    });
+            CustomApplicationUtils.modifyEndpointForCustomApplication(config, application);
```

Comment:
The result is not used. So it does nothing.

Reply:
fixed